src/tasks/WasmBuildTasks/LinkContentToWwwroot.cs

```diff
@@ -0,0 +1,120 @@
+                contentItems.Add(outItem);
+
+                Log.LogMessage(MessageImportance.Low, $"Adding TargetPath '{targetPath}' and ContentRoot '{contentRoot}' for Identity '{identity}'.");
+                continue;
+            }
+
+            // Case 2: use Identity when Link is empty
+            if (copyPreserveOrAlways && string.IsNullOrEmpty(link))
+            {
+                var isRooted = Path.IsPathRooted(identity);
+                targetPath = isRooted ? Path.GetFileName(identity) : identity;
+                var contentRoot = isRooted ? Path.GetDirectoryName(identity) : Path.GetDirectoryName(Path.GetFullPath(identity, MSBuildProjectDirectory));
+
+                // Add TargetPath to WasmFilesToIncludeInFileSystem
+                var wasmVfsFile = new TaskItem(targetPath);
+                wasmFiles.Add(wasmVfsFile);
+                if (isRooted && identity.StartsWith(MSBuildProjectDirectory, StringComparison.OrdinalIgnoreCase))
+                {
+                    var wasmVfsTargetPath = Path.GetRelativePath(MSBuildProjectDirectory, identity);
+                    if (!string.IsNullOrEmpty(wasmVfsTargetPath))
+                        wasmVfsFile.SetMetadata("TargetPath", wasmVfsTargetPath);
+                }
+
+                var outItem = new TaskItem(identity, item.CloneCustomMetadata());
+                outItem.SetMetadata("ContentRoot", contentRoot);
+                outItem.SetMetadata("TargetPath", Path.Combine("wwwroot", targetPath));
+                contentItems.Add(outItem);
+
+                Log.LogMessage(MessageImportance.Low, $"Adding TargetPath '{targetPath}' and ContentRoot '{contentRoot}' for Identity '{identity}'.");
+                continue;
+            }
+
+            // Case 3: update Link to point to wwwroot
+            if (copyPreserveOrAlways && !string.IsNullOrEmpty(link) && !link.StartsWith("wwwroot"))
+            {
+                var isRooted = Path.IsPathRooted(identity);
+                var contentRoot = isRooted ? Path.GetDirectoryName(identity) : Path.GetDirectoryName(Path.GetFullPath(identity, MSBuildProjectDirectory));
+
+                // Add Link to WasmFilesToIncludeInFileSystem
+                wasmFiles.Add(new TaskItem(link));
+
+                var outItem = new TaskItem(identity, item.CloneCustomMetadata());
+                outItem.SetMetadata("ContentRoot", contentRoot);
+                outItem.SetMetadata("Link", Path.Combine("wwwroot", link));
+                contentItems.Add(outItem);
+
+                Log.LogMessage(MessageImportance.Low, $"Adding Link '{link}' and ContentRoot '{contentRoot}' for Identity '{identity}'.");
+                continue;
+            }
+
+            Log.LogMessage(MessageImportance.Low, $"Skipping item with Identity '{identity}' (TargetPath '{targetPath}', Link '{link}') because CopyToOutputDirectory is not 'PreserveNewest' or 'Always' ('{copyPreserveOrAlways}').");
```

Comment:
Nit: could we change the order? First check `if(!copyPreserveOrAlways)` -> log skip message + `continue`, then drop checking that bool in each following condition and make them if-else-else?